go/rows.go

```diff
@@ -121,6 +124,13 @@ func (r *Rows) Next(dest []driver.Value) error {
 	if r.reachedLastPage {
 		return io.EOF
 	}
+
+	if r.statistics != nil && len(dest) > 0 {
```

Comment:
could you check for len(dest) == 1 & dest[0] type is QueryStatsNil? Then there would probably be no need to clear the r.statistics afterwards.

Reply:
It will not work. The length of `dest` is always the same as number of columns in the original query. If the query returns 9 columns the len of 'dest' is 9 and I add the metrics to the first column, I do not fill the following 8 . In g2 backend I have to scan 9 columns `rows.Scan([9]values)` to get metrics